streams/src/main/java/org/apache/kafka/streams/processor/internals/ProcessorContextImpl.java

```diff
@@ -118,16 +111,24 @@ public void logChange(final String storeName,
                           final byte[] value,
                           final long timestamp) {
         throwUnsupportedOperationExceptionIfStandby("logChange");
+
+        final TopicPartition changelogPartition = stateManager().changelogTopicPartitionFor(storeName);
+        if (changelogPartition == null) {
+            throw new IllegalStateException("Sending records to state store " + storeName +
+                " which has not been registered.");
```

Comment:
I thinks the store could still be registered, but just does not have a changelog topic? Wondering if the error message might be miss leading?

Reply:
Good catch! I will add the logging disabled case to the message.